bibliopixel/drivers/SPI/base.py

```diff
@@ -0,0 +1,27 @@
+from .. channel_order import ChannelOrder
+from .. driver_base import DriverBase
+from ... import gamma as _gamma
+from ... util.enum import resolve_enum
+from . import interfaces
+
+
+class SPIBase(DriverBase):
+    """Base driver for controling SPI devices on systems like the Raspberry Pi and BeagleBone"""
+    def __init__(self, num, dev='/dev/spidev0.0',
+                 interface='FILE', spi_speed=1,
```

Comment:
As much as possible I want every default option to be the one that has no extra dependencies. With the exception of WS281X, every driver works out of the box with file mode. Maybe not as well, but good enough for most people. I could maybe add an warning message every time you use file mode letting them know that something better exists. But keeping it file mode maintains our desire to have it just work out of the box.